readersampleprovider.go

```diff
@@ -71,6 +73,12 @@ func ReaderTrackWithRTCPHandler(f func(rtcp.Packet)) func(provider *ReaderSample
 	}
 }
 
+func ReaderTrackDisableAutoClose() func(provider *ReaderSampleProvider) {
```

Comment:
should it always set this? would be good to ensure we can always correctly republish upon reconnect.

Reply:
It requires the application to call close to release resources if it is enabled as we can't release resources in unbind method. Concerning it would cause resource leak in current applications if we enable it by default.

Reply:
right.. I think the alternatives is that reconnect would produce failures? it seems neither path is ideal.

could Room or Engine close published tracks upon disconnect? that's the path other client SDKs take.

Reply:
Thought of this way, but the go-sdk only takes webrtc.TrackLocal interface in `PublishTrack` method that don't have a `Close` method. 
A good way would be to output a error log on republish failure that notify the users to use the new API(new option, and Close method to clean resource). In this way, the application will not have loss after updated the sdk, no resource leak, and no republish improvement but has a message to tell him the right way. Or any other suggestions?

Reply:
We could have another interface like `TrackLocalWithClose` ? that extends `webrtc.TrackLocal`? and as you suggested:
* if the track isn't a `TrackLocalWithClose`, then log warning so dev remembers to cleanup
* if the track is a `TrackLocalWithClose`, then cleanup automatically.

Reply:
actually, we might not even require a warning.. since the user is responsible for handling cleaning up in Unbind.. and when republished, Bind will be called again.

Reply:
The problem is  if the resource is cleaned in the unbind, bind will failed as the reader is closed.

Reply:
> if the track isn't a TrackLocalWithClose, then log warning so dev remembers to cleanup
if the track is a TrackLocalWithClose, then cleanup automatically.

Sounds good, I'll try it